Jetpack Manage: Restructure the navigator back button to separate the title and icon.

jkguidaven opened the pull request:
As the core design team recommended, we should separate the back button and the title of the navigation menu. This PR fixes this discrepancy. 

https://github.com/Automattic/wp-calypso/assets/56598660/7d2cc9a5-a720-4bb8-90df-8cce96426641


Closes https://github.com/Automattic/jetpack-genesis/issues/84

## Proposed Changes

* Add another **'title'** prop to the menu item and convert the **'label'** props of the back button as a tooltip text.
* Update consumers accordingly with the new changes.

## Testing Instructions

**Prerequisites**

Since these changes are made specifically for agencies, you must set yourself(partner) as an agency - 2c49b-pb. Make sure to switch it back to the previous type.

* Use the Jetpack Cloud live link and go to `/partner-portal/billing`
* Confirm that the back button is now a separate element to the menu title. Ensure the hovering tooltip works as expected.
<img width="271" alt="Screen Shot 2023-10-31 at 1 54 49 PM" src="https://github.com/Automattic/wp-calypso/assets/56598660/7fc0a932-bdde-4a45-88f5-5ecfe0ea5486">

* Confirm also that the menu title is changed in the single site view to **'Manage'**.
<img width="272" alt="Screen Shot 2023-10-31 at 1 54 58 PM" src="https://github.com/Automattic/wp-calypso/assets/56598660/4f981690-24e3-4944-bab8-574b2203f4f0">

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

jkguidaven committed:
Restructure navigation back button to separate title and icon.

jkguidaven committed:
Allow title to be displayed separately with back button.

jkguidaven commented on client/layout/sidebar-v2/navigator/navigator-menu.tsx:
```diff
@@ -9,38 +9,50 @@ import {
 							) }
-							<div>
+
+							<li>
```
Lint complains that only `li` can be a direct child to `ul`.

ilonagl reviewed (approved):
It looks great. Thank you!

yashwin reviewed (commented):
Thanks for working on this, @jkguidaven!

I think we should still keep the title as `Sites` as discussed here: 
p1HpG7-ps9-p2#comment-67031

@ilonagl: What do you think?

ilonagl commented:
> @ilonagl: What do you think?

Agreed, sorry for missing this during my review.

yashwin commented:
> > p1HpG7-ps9-p2#comment-67031
> 
> @yashwin and @ilonagl If I recall, the reason why we switched to renaming it to `Sites` was to make sure we are informing the user that pressing the label will return back to the `Sites` page, but now that the title is a separate element. Is this still the case?

I think so. The reason is: We have just changed the way it looks internally. It is still the same for users look-wise.

jkguidaven commented:
> I think so. The reason is: We have just changed the way it looks internally. It is still the same for users look-wise.

Sorry, I do not see the point of separating the element if the purpose of the title is not to describe the entire menu view. Or maybe I'm missing something here. 🤔

I am reading the conversation here. I am under the impression we did the rename because of how initially the chevron and label were a single component, but now that we have changed it. It shouldn't be the case, right?

p1HpG7-ps9-p2#comment-67030

yashwin commented:
> Sorry, I do not see the point of separating the element if the purpose of the title is not to describe the entire menu view. Or maybe I'm missing something here. 🤔

I'm not sure if this was an immediate requirement, and I feel we should discuss this and come to a conclusion as mentioned here: p1HpG7-ps9-p2#comment-67060 

As per @ilonagl's comment: https://github.com/Automattic/wp-calypso/pull/83510#issuecomment-1780866119, it is suggested to keep it as it is for now. So, I think we should continue discussing this further and keep this PR on hold for now 🙂 

WDYT?

jkguidaven commented:
> > Sorry, I do not see the point of separating the element if the purpose of the title is not to describe the entire menu view. Or maybe I'm missing something here. 🤔
> 
> I'm not sure if this was an immediate requirement, and I feel we should discuss this and come to a conclusion as mentioned here: p1HpG7-ps9-p2#comment-67060
> 
> As per @ilonagl's comment: https://github.com/Automattic/wp-calypso/pull/83510#issuecomment-1780866119, it is suggested to keep it as it is for now. So, I think we should continue discussing this further and keep this PR on hold for now 🙂
> 
> WDYT?

Make sense.  🙂 👍 I agree this PR is not urgent and can wait.

ilonagl commented:
>  So, I think we should continue discussing this further and keep this PR on hold for now 🙂

If we could do this, it would be the best solution. I'll gather the design team's thoughts on this in the meanwhile. 

My apologies for this back and forth on this. It's a tricky one.

yashwin commented:
> If we could do this, it would be the best solution. I'll gather the design team's thoughts on this in the meanwhile.

Thank you! 🙂 

> My apologies for this back and forth on this. It's a tricky one.

No worries at all. Please take your time 🙂 Yes, it is a trick one 👍

ilonagl commented:
Hey @jkguidaven @yashwin, 

I just discussed this with the design team and the agreed direction is to keep our designs in-sync with the core's team. That said, let's push this PR forward.

ilonagl commented:
For the namings: let's use "Purchases" and for the cloud view "Manage". cc @jeffgolenski

jkguidaven committed:
Rename 'Site Settings' to 'Manage'.

jkguidaven committed:
Hide sidebar title on single site view for non-agency users.

jkguidaven committed:
Remove focusing box shadow for menu items.

yashwin reviewed (approved):
It works as described in the instructions.  Thanks for working on this. 

Also, could we please add the screenshots to make it easy for the translators?
